jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala

```diff
@@ -0,0 +1,38 @@
+package ml4ir.inference.tensorflow.data
+
+import java.io.File
+
+import com.fasterxml.jackson.annotation.{JsonIgnoreProperties, JsonProperty}
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.google.common.base.Charsets._
+import com.google.common.io.Files
+import org.tensorflow.DataType
+
+import scala.collection.JavaConverters._
+
+object ModelFeaturesConfig {
+  def load(configPath: String): ModelFeaturesConfig = {
+    val objectMapper = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    objectMapper.registerModule(DefaultScalaModule)
+    objectMapper.readValue[ModelFeaturesConfig](Files.readLines(new File(configPath), UTF_8).asScala.mkString("\n"))
+  }
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+case class ModelFeaturesConfig(@JsonProperty("rank") initialRank: FeatureConfig,
+                               @JsonProperty("features") features: List[FeatureConfig])
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+case class FeatureConfig(@JsonProperty("node_name") nodeName: String,
+                         @JsonProperty("dtype") dTypeString: String,
+                         @JsonProperty("serving_info") servingConfig: ServingConfig,
+                         @JsonProperty("default_value") defaultValue: String,
```

Comment:
With this, we are NOT moving default_value under serving_info. Is this something we stick with for now? We can update it in v2 but this will have a cascading effect later on (i.e. changes to yaml and backward compatibility).

Reply:
ah, we can do either way - what did we say we wanted to do?  I can change it to whatever way we want the config to be.  I didn't want to break any current configs.

We don't need to worry about backwards compatibility until this ships, and even then, until a major version of ml4ir ships, we can change, as long as we know models go out with new jars as needed.

Reply:
Also update the model_features.yaml to move the default_value under serving_info.